src/analysis/normed/ring/seminorm.lean

```diff
@@ -278,7 +280,83 @@ instance : has_one (mul_ring_norm R) :=
 
 instance : inhabited (mul_ring_norm R) := ⟨1⟩
 
+lemma map_neg_one {R : Type*} [non_assoc_ring R] {f : mul_ring_norm R} :
+  f (-1) = 1 :=
+by rw [map_neg_eq_map, map_one]
+
+/-- Two multiplicative ring norms `f, g` on `R` are equivalent if there exists a positive constant
+  `c` such that for all `x ∈ R`, `(f x)^c = g x`. -/
+def equiv {R : Type*} [ring R] (f g : mul_ring_norm R) :=
+  ∃ c : ℝ, 0 < c ∧ (λ x : R, (f x) ^ c) = g
+
+lemma equiv.refl {R : Type*} [ring R] (f : mul_ring_norm R) :
+  equiv f f := by refine ⟨1, by linarith, by simp only [real.rpow_one]⟩
+
+lemma equiv.symm {R : Type*} [ring R] {f g : mul_ring_norm R} (hfg : equiv f g) :
+  equiv g f :=
+begin
+  obtain ⟨c, hc, hfg⟩ := hfg,
+  refine ⟨c⁻¹, inv_pos.mpr hc, funext $ λ x, _⟩,
+  rw [← hfg, ←real.rpow_mul (map_nonneg f x), mul_inv_cancel hc.ne', real.rpow_one],
+end
+
+lemma equiv.trans {R : Type*} [ring R] (f g k : mul_ring_norm R)
+  (hfg : equiv f g) (hgk : equiv g k) : equiv f k :=
+begin
+  rcases hfg with ⟨c, hfg1, hfg2⟩,
+  rcases hgk with ⟨d, hgk1, hgk2⟩,
+  refine ⟨c * d, by simp only [hfg1, hgk1, zero_lt_mul_right], _⟩,
+  rw [← hgk2, ← hfg2],
+  ext,
+  exact real.rpow_mul (map_nonneg f x) c d,
+end
+
+end mul_ring_norm
+
+section nonarchimedean_mul_ring_norm
+
+/-- A function `f : α → β` is nonarchimedean if it satisfies the inequality
+  `f (a + b) ≤ max (f a) (f b)` for all `a, b ∈ α`. -/
+def is_nonarchimedean {α : Type*} [has_add α] {β : Type*} [linear_order β] (f : α → β) : Prop :=
```

Comment:
I feel a bit confused here. Since we use `ring` and `ring_seminorm_class` for the lemmas here, why should we move it to the group section?

Reply:
You do not need ring or ring_seminorm class for `is_nonarchimedean` and `is_nonarchimedean_def`, and this notion also makes sense for seminorms on additive groups.

Reply:
ok, I got that. But when I want to import `ring_norm_class` from `analysis.normed.ring.seminorm` in `analysis.normed.group.seminorm`, my lean just broke :(

Reply:
It broke because if you do that you are creating circular imports, but that is not what I am suggesting to do. You should move `is_nonarchimedean` and `is_nonarchimedean_def` to `analysis.normed.group.seminorm` (which doesn't require any extra imports). I am not suggesting that you move anything else there (because the other lemmas do use `ring_seminorm_class`, so they should stay in `analysis.normed.ring.seminorm`).

Reply:
ok，I got that，sorry for misreading